x-pack/plugins/cases/server/client/attachments/bulk_delete.ts

```diff
@@ -0,0 +1,159 @@
+import type { BulkDeleteFileArgs } from './types';
+import { constructOwnerFromFileKind, CaseFileMetadataForDeletionRt } from '../../../common/files';
+import type { CasesClient } from '../client';
+
+export const bulkDeleteFileAttachments = async (
+  { caseId, fileIds }: BulkDeleteFileArgs,
+  clientArgs: CasesClientArgs,
+  casesClient: CasesClient
+) => {
+  const {
+    user,
+    services: { attachmentService, userActionService },
+    logger,
+    authorization,
+    fileService,
+  } = clientArgs;
+
+  try {
+    const request = pipe(
+      excess(BulkDeleteFileAttachmentsRequestRt).decode({ ids: fileIds }),
+      fold(throwErrors(Boom.badRequest), identity)
+    );
+
+    await casesClient.cases.resolve({ id: caseId, includeComments: false });
+
+    const fileEntities = await getFileEntities(caseId, request.ids, fileService);
+
+    // It's possible for this to return an empty array if there was an error creating file attachments in which case the
+    // file would be present but the case attachment would not
+    const fileAttachments = await attachmentService.getter.getFileAttachments({
+      caseId,
+      fileIds: request.ids,
+    });
+
+    await authorization.ensureAuthorized({
+      entities: [
+        ...fileAttachments.map((attachment) => ({
+          id: attachment.id,
+          owner: attachment.attributes.owner,
+        })),
+        ...fileEntities,
+      ],
+      operation: Operations.deleteComment,
+    });
+
+    await Promise.all([
+      pMap(request.ids, async (fileId: string) => fileService.delete({ id: fileId }), {
+        concurrency: MAX_CONCURRENT_SEARCHES,
+      }),
+      attachmentService.bulkDelete({
+        attachmentIds: fileAttachments.map((so) => so.id),
+        refresh: false,
+      }),
+    ]);
+
+    await userActionService.creator.bulkCreateAttachmentDeletion({
+      caseId,
+      attachments: fileAttachments.map((attachment) => ({
+        id: attachment.id,
+        owner: attachment.attributes.owner,
+        attachment: attachment.attributes,
+      })),
+      user,
+    });
+  } catch (error) {
+    throw createCaseError({
+      message: `Failed to delete file attachments for case: ${caseId}: ${error}`,
+      error,
+      logger,
+    });
+  }
+};
+
+const getFileEntities = async (
+  caseId: BulkDeleteFileArgs['caseId'],
+  fileIds: BulkDeleteFileArgs['fileIds'],
+  fileService: FileServiceStart
+) => {
+  const files = await getFiles(caseId, fileIds, fileService);
+
+  const fileEntities = createFileEntities(files);
+
+  return fileEntities;
+};
+
+const getFiles = async (
+  caseId: BulkDeleteFileArgs['caseId'],
+  fileIds: BulkDeleteFileArgs['fileIds'],
+  fileService: FileServiceStart
+) => {
+  // it's possible that we're trying to delete a file when an attachment wasn't created (for example if the create
+  // attachment request failed)
+  const files = await pMap(fileIds, async (fileId: string) => fileService.getById({ id: fileId }), {
```

Comment:
@elastic/appex-sharedux Is it possible to expose a bulk get operation to improve performance?